### ISSUE TITLE
unknown field ExcessDataGas in struct literal

### ISSUE THREAD

**Issue by Mixnosha** [zksync-sdk/zksync2-go#27]
go.mod

```
module l2

go 1.21.3

require (
	github.com/ethereum/go-ethereum v1.13.8
	github.com/zksync-sdk/zksync2-go v0.3.2
)
```

Code from example 

```
package main

import (
	"github.com/zksync-sdk/zksync2-go/clients"
)

func main() {
	ZkSyncEraProvider := "https://testnet.era.zksync.dev"
	ZkSyncEraWSProvider := "ws://testnet.era.zksync.dev:3051"

	// Connect to zkSync network
	client, err := clients.Dial(ZkSyncEraProvider)
	if err != nil {
		panic(err)
	}
	defer client.Close()

	// Connect to zkSync network using Web Socket
	wsClient, err := clients.Dial(ZkSyncEraWSProvider)
	if err != nil {
		panic(err)
	}
	defer wsClient.Close()

}

```

Command ``` go run main.go```

Error 

```
# github.com/zksync-sdk/zksync2-go/clients
../../../../go/pkg/mod/github.com/zksync-sdk/zksync2-go@v0.3.2/clients/base_client.go:743:4: unknown field ExcessDataGas in struct literal of type "github.com/ethereum/go-ethereum/core/types".Header
```

**Comment by danijelTxFusion** [comment]
Version `0.3.2` does not work with `geth: 1.13`, you need to downgrade to `1.12.0`. We are currently migrating to the latest version of `geth`, so the new version should be released soon.

**Comment by danijelTxFusion** [comment]
Also WebSocket connection are not allowed on testnet.

**Comment by yibishufange** [comment]
> Also WebSocket connections are not allowed on testnet.

which means i cannot subscribe to contract events via ws connections?

**Comment by danijelTxFusion** [comment]
> > Also WebSocket connections are not allowed on testnet.
> 
> which means i cannot subscribe to contract events via ws connections?

Correct. On testnet node is overloaded with traffic and listening on events is not possible.

**Comment by danijelTxFusion** [comment]
It should be done in 2 days.

**Comment by danijelTxFusion** [comment]
Fixed in `v0.4.0`.
